base/range.jl

```diff
@@ -374,7 +374,7 @@ julia> step(range(2.5, stop=10.9, length=85))
 ```
 """
 step(r::StepRange) = r.step
-step(r::AbstractUnitRange) = 1
+step(r::AbstractUnitRange{T}) where{T} = oneunit(T)
```

Comment:
I think this is a good change, but could we add a `::T` assertion here to ensure that we're in line with the step type we report to `OrdinalRange`?